apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -1,267 +1,394 @@
+        assert(scannedImagePaths);
+        assert(electionDefinition);
 
-    await enablePrintPromise;
-    let dotsSkipped = 0;
-    debug(`begin printing ${customChunkedBitmaps.length} chunks`);
-    let i = 0;
-    for (const customChunkedBitmap of customChunkedBitmaps) {
-      debug(`printing chunk ${i}`);
-      if (customChunkedBitmap.empty) {
-        dotsSkipped += VERTICAL_DOTS_IN_CHUNK;
-      } else {
-        if (dotsSkipped) {
-          await this.driver.setRelativeVerticalPrintPosition(dotsSkipped * 2); // assuming default vertical units, 1 / 408 units
-          dotsSkipped = 0;
-        }
-        await this.driver.printChunk(customChunkedBitmap);
-      }
-      i += 1;
-    }
-    debug('done printing ballot');
-  }
-}
+        const sheetId = uuid();
+        const { precincts } = electionDefinition.election;
+        // Hard coded for now because we don't store precinct in backend. This
+        // will be replaced with a store read in a future PR.
+        const precinct = precincts[precincts.length - 1];
+        const precinctSelection = singlePrecinctSelectionFor(precinct.id);
+        const testMode = true;
+        return interpretScannedBallots(
+          electionDefinition,
+          precinctSelection,
+          testMode,
+          scannedImagePaths,
+          sheetId
+        );
+      },
+      onDone: {
+        target: 'presenting_ballot',
+      },
+    },
+  ],
+};
 
-export async function getPaperHandlerStateMachine(): Promise<
-  Optional<PaperHandlerStateMachine>
+const PresentingBallotState: StateNodeConfig<
+  Context,
+  any,
+  PaperHandlerStatusEvent,
+  BaseActionObject
+> = {
+  entry: async (context) => {
+    await context.driver.presentPaper();
+  },
+};
+
+const FinalState: StateNodeConfig<
+  Context,
+  any,
+  PaperHandlerStatusEvent,
+  BaseActionObject
+> = {
+  type: 'final',
+  entry: () => {
+    debug('Entered ending state');
+  },
+};
+
+export function buildMachine(
+  workspace: Workspace,
+  driver: PaperHandlerDriver,
+  pollingIntervalMs: number
+): StateMachine<
+  Context,
+  StateSchema,
+  PaperHandlerStatusEvent,
+  any,
+  BaseActionObject,
+  ServiceMap
 > {
-  const paperHandlerDriver = await getPaperHandlerDriver();
-  if (!paperHandlerDriver) return;
+  return createMachine({
+    schema: {
+      /* eslint-disable-next-line vx/gts-object-literal-types */
+      context: {} as Context,
+      /* eslint-disable-next-line vx/gts-object-literal-types */
+      events: {} as PaperHandlerStatusEvent,
+    },
+    id: 'bmd',
+    initial: 'no_paper',
+    context: {
+      workspace,
+      driver,
+      pollingIntervalMs,
+    },
+    states: {
+      no_paper: NoPaperState,
+      loading_paper: LoadingPaperState,
+      waiting_for_ballot_data: WaitingForBallotDataState,
+      printing_ballot: PrintingBallotState,
+      scanning: ScanningState,
+      interpreting: InterpretingState,
+      presenting_ballot: PresentingBallotState,
+      final_state: FinalState,
```

Comment:
What do you think about inlining the states? I think it's a little easier to read when you can easily find the state name referenced by another state in the same object.

Reply:
I saw `scan` does that and was split on approach for `mark-scan`. On one hand I like being able to see all the high level states at a glance, something that's hard to do with `scan`'s inlined states, but on the other hand

* it's annoying and error-prone to keep the separate state declarations in order
* I agree it's easier to do an in-depth read of the code when they're inlined

I think the pros of inlining outweigh the cons so I'll update.